src/webhook/mutation/pod_mutator/oneagent_mutation/versionlabel.go

```diff
@@ -0,0 +1,29 @@
+package oneagent_mutation
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	defaultVersionLabelMapping = VersionLabelMapping{
+		releaseVersionEnv: "metadata.labels['app.kubernetes.io/version']",
+		releaseProductEnv: "metadata.labels['app.kubernetes.io/part-of']",
+	}
+)
+
+type VersionLabelMapping map[string]string
+
+func newVersionLabelMapping(namespace corev1.Namespace) VersionLabelMapping {
+	// TODO
```

Comment:
please add a notice when these TODOs can be removed, so they do not stick around